pydatalab/src/pydatalab/routes/v0_1/items.py

```diff
@@ -989,6 +1004,345 @@ def get_item_data(item_id: str | None = None, refcode: str | None = None):
+
+    # Protect critical fields from being overwritten during restore
+    protected_fields = [
+        "refcode",
+        "_id",
+        "immutable_id",
+        "creator_ids",
+        "file_ObjectIds",
+        "version",
+    ]
+    for field in protected_fields:
+        if field in current_item:
+            restored_data[field] = current_item[field]
+
+    # Ensure type consistency
+    if restored_data.get("type") != current_item.get("type"):
+        return jsonify(
+            {
+                "status": "error",
+                "message": f"Cannot restore version with different type. Current: {current_item.get('type')}, Version: {restored_data.get('type')}",
+            }
+        ), 400
+
+    # Atomically get the next version number (used for both version_number and item.version)
+    next_version_number = _get_next_version_number(refcode)
+
+    # Update metadata to reflect the restore action
+    restored_data["version"] = next_version_number
+    restored_data["last_modified"] = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+
+    # Validate restored data against the item model
+    item_type = current_item["type"]
+    if item_type not in ITEM_MODELS:
+        return jsonify({"status": "error", "message": f"Invalid item type: {item_type}"}), 400
+
+    try:
+        # Validate using the appropriate model
+        ITEM_MODELS[item_type](**restored_data)
+    except ValidationError as exc:
+        return jsonify(
+            {
+                "status": "error",
+                "message": f"Restored data failed validation: {str(exc)}",
+                "output": str(exc),
+            }
+        ), 400
+
+    # Perform the restore first
+    flask_mongo.db.items.update_one({"refcode": refcode}, {"$set": restored_data})
+
+    # Extract user information for hybrid storage approach
+    user_snapshot = None
+    user_id = None
+    if current_user.is_authenticated:
+        user_id = current_user.person.immutable_id
+        user_snapshot = {
+            "id": str(user_id),
+            "display_name": getattr(current_user.person, "display_name", None),
+            "email": getattr(current_user.person, "contact_email", None),
+        }
+
+    # Get the software version
+    try:
+        software_version = get_package_version("datalab-server")
+    except PackageNotFoundError:
+        software_version = "unknown"
+
+    # Save the RESTORED state as a new version snapshot (after restore)
+    flask_mongo.db.item_versions.insert_one(
+        {
+            "refcode": refcode,
+            "version_number": next_version_number,
+            "timestamp": datetime.datetime.now(tz=datetime.timezone.utc).isoformat(),
+            "action": "restored",  # Audit trail: this is a restored version
+            "restored_from_version": str(
+                version_object_id
+            ),  # Track which version was restored from
+            "user": user_snapshot,  # Snapshot for fast display
```

Comment:
```suggestion
```

As mentioned, I'd just store the ID then recreate it on egress via something like the `creators_lookup` method in this file which does an aggregation as:

```python
def creators_lookup() -> dict:
    return {
        "from": "users",
        "let": {"creator_ids": "$creator_ids"},
        "pipeline": [
            {"$match": {"$expr": {"$in": ["$_id", {"$ifNull": ["$$creator_ids", []]}]}}},
            {"$addFields": {"__order": {"$indexOfArray": ["$$creator_ids", "$_id"]}}},
            {"$sort": {"__order": 1}},
            {"$project": {"_id": 1, "display_name": 1, "contact_email": 1}},
        ],
        "as": "creators",
    }
```